samples/bluetooth/central_uart/src/main.c

```diff
@@ -553,11 +561,13 @@ static struct bt_conn_auth_info_cb conn_auth_info_callbacks = {
 	.pairing_complete = pairing_complete,
 	.pairing_failed = pairing_failed
 };
+#endif
 
 int main(void)
 {
 	int err;
 
+#ifdef CONFIG_BT_NUS_SECURITY_ENABLED
```

Comment:
Please try to use IS_ENABLED macro like it is done in the peripheral_uart sample